### PR TITLE
[huggingface_tensorflow] Update Framework version to TF2.8

### PR DESCRIPTION
**Pull request by philschmid**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: huggingface/tensorflow/inference/docker/2.8/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    LD_LIBRARY_PATH="/opt/conda/lib/:${LD_LIBRARY_PATH}:/usr/local/lib" \
+    PYTHONIOENCODING=UTF-8 \
+    LANG=C.UTF-8 \
+    LC_ALL=C.UTF-8 \
+    TEMP=/home/model-server/tmp \
+    DEBIAN_FRONTEND=noninteractive
+
+ENV PATH /opt/conda/bin:$PATH
+
+RUN apt-get update \
```
Comment (arjkesh) [inline huggingface/tensorflow/inference/docker/2.8/py3/Dockerfile.cpu:32]: it looks like something in this statement is returning nonzero exit code. can you debug, either by reproducing manually, or by splitting the statement into multiple run commands to identify where it is failing?

Comment (philschmid) [reply huggingface/tensorflow/inference/docker/2.8/py3/Dockerfile.cpu:32]: I changed it to individual installs. But i cannot check the logs since it is "stuck". P.S. Its the same packages we use PyTorch CPU Inference container so not sure why this should fail

**Comment by kevinyang8** [comment]
@philschmid Please refer to the comment I left on this [PR](https://github.com/aws/deep-learning-containers/pull/1870) about test_anaconda. You can reference changes I made there for removing installations from repo.anaconda.com. These will need to be made for any future DLC releases and we are working on informing our partner teams.

File: huggingface/tensorflow/inference/docker/2.8/py3/Dockerfile.cpu (hunk 1 above)
Comment (shantanutrip) [reply huggingface/tensorflow/inference/docker/2.8/py3/Dockerfile.cpu:32]: On reading the logs I found out that GPU images , training and inference both, succeed the image size check. However, we do not see anything about the CPU image size check. There is a probability that CPU inference image fails the size check and exits in a way that causes the hang. I would suggest increasing the image baseline size for the CPU inference image and see if the builds succeed.

Comment (shantanutrip) [reply huggingface/tensorflow/inference/docker/2.8/py3/Dockerfile.cpu:32]: @philschmid I think that was the reason. The builds have started succeeding. We would want to take the following steps now:
1. See what is the new size of the CPU inference images and what was the old size. If there is a drastic difference, we might want to dive deep and see what is causing the inflation. 
2. As of now, I have set the `image_size_baseline` to 6000 (that I increased from 3000). We would want to set this size based on the size of the newly built images.

File: huggingface/tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+ && ./configure --prefix=/home/.openmpi \
+ && make all install \
+ && cd .. \
+ && rm openmpi-$OPEN_MPI_VERSION.tar.gz \
+ && rm -rf openmpi-$OPEN_MPI_VERSION
+
+
+ENV PATH="$PATH:/home/.openmpi/bin"
+ENV LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/home/.openmpi/lib/"
+RUN ompi_info --parsable --all | grep mpi_built_with_cuda_support:value
```
Comment (shantanutrip) [inline huggingface/tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu:64]: Error seen here:

```


[ERROR] ERROR during Docker BUILD
--
398 | [ERROR] Error message received for docker/2.8/py3/cu112/Dockerfile.gpu while docker build: {'errorDetail': {'code': 1, 'message': "The command '/bin/sh -c ompi_info --parsable --all \| grep mpi_built_with_cuda_support:value' returned a non-zero code: 1"}, 'error': "The command '/bin/sh -c ompi_info --parsable --all \| grep mpi_built_with_cuda_support:value' returned a non-zero code: 1"}
```

**Review by shantanutrip — COMMENTED** [review]
Hey @philschmid , can you please resolve the conflicts and merge the master into this branch once again?

**Comment by tejaschumbalkar** [comment]
@philschmid Can you close the PR if it is not valid or move it to draft if it is not being worked currently?
